docker/proto/Dockerfile

```diff
@@ -0,0 +1,38 @@
+FROM golang:1.14-buster
+
+# Install depedencies
+RUN apt-get update && \
+    DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+        build-essential \
+        ca-certificates \
+        curl \
+        unzip \
+        wget
+
+ARG PROTOC_GEN_GO_VERSION=v1.3.2
+
+# Need to set up a module in order to install a specific version of protoc-gen-go.
+#   1. The `go get package@version` syntax is only supported in module mode.
+#   2. In module mode, `go get` is unsupported outside of a module.
+# These two things together force us to do this quick hack.
+#
+# See https://github.com/golang/go/issues/24250 as just one of many issues on the subject.
+RUN mkdir proto-builder && \
+    cd proto-builder && \
+    go mod init proto-builder && \
+    GO111MODULE=on go get \
+        golang.org/x/tools/cmd/goimports \
+        github.com/golang/protobuf/protoc-gen-go@${PROTOC_GEN_GO_VERSION}
+
+ARG PROTOC_VERSION=3.12.4
+
+# Download protoc
+RUN curl -LO https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOC_VERSION}/protoc-${PROTOC_VERSION}-linux-x86_64.zip && \
+    unzip protoc-${PROTOC_VERSION}-linux-x86_64.zip -d /protoc && \
+    mv /protoc/bin/protoc /usr/local/bin/ && \
+    rm -rf protoc-${PROTOC_VERSION}-linux-x86_64.zip
```

Comment:
I am not sure what the end use of this container would be, but I would support having it be built once and pushed to the vitess repo on docker hub. That would prevent the fetching of these dependencies every time we rebuild proto's

Reply:
Pushing this up somewhere sounds like a great idea! Do you think it should go under vitess's dockerhub, or somewhere else?